.github/workflows/quality.yaml

```diff
@@ -24,7 +24,10 @@ jobs:
       - name: Install dependencies
         run: |
           python -m pip install --upgrade pip
-          python -m pip install ruff==v0.1.2
+          python setup.py egg_info
```

Comment:
This new mechanism 
1) builds a requirement.txt file
2) extracts the mandatory requirements
3) installs ruff using the version in teh mandatory requirements

Reply:
(`-c` = constraint file in pip)